src/js/port_handler.js

```diff
@@ -44,6 +142,10 @@ PortHandler.check = function () {
         self.check_serial_devices();
     }
 
+    self.check_elrs_devices();
```

Comment:
Where is this function defined?

Reply:
Making this changes restores PortHandler to normal operation. Did not test ELRS functionality:

```
diff --git a/src/js/port_handler.js b/src/js/port_handler.js
index a9230135..dab965bb 100644
--- a/src/js/port_handler.js
+++ b/src/js/port_handler.js
@@ -1,40 +1,5 @@
 'use strict';
 
-const http = require("http");
-const bonjour = require('bonjour')();
-
-const MDNS_INTERVAL = 10000;
-const TCP_CHECK_INTERVAL = 5000;
-const TCP_TIMEOUT = 2000;
-
-let mdnsBrowser = null;
-if (GUI.isCordova()) {
-    mdnsBrowser = {
-        services: [],
-    };
-
-    const zeroconf = cordova.plugins.zeroconf;
-    zeroconf.registerAddressFamily = 'ipv4'; // or 'ipv6' ('any' by default)
-    zeroconf.watchAddressFamily = 'ipv4'; // or 'ipv6' ('any' by default)
-    zeroconf.watch("_http._tcp.", "local.").subscribe(result => {
-        const action = result.action;
-        const service = result.service;
-        if (['added', 'resolved'].includes(action)) {
-            console.log("Zeroconf Service Changed", service);
-            mdnsBrowser.services.push({
-                addresses: service.ipv4Addresses,
-                txt: service.txtRecord,
-                fqdn: service.hostname,
-            });
-        } else {
-            console.log("Zeroconf Service Removed", service);
-            mdnsBrowser.services = mdnsBrowser.services.filter(s => s.fqdn !== service.hostname);
-        }
-    });
-} else {
-    bonjour.find({ type: 'http' });
-}
-
 const TIMEOUT_CHECK = 500 ; // With 250 it seems that it produces a memory leak and slowdown in some versions, reason unknown
 
 const usbDevices = { filters: [
@@ -50,6 +15,7 @@ const PortHandler = new function () {
     this.port_available = false;
     this.showAllSerialDevices = false;
     this.showVirtualMode = false;
+    this.mdnsBrowser = null;
 };
 
 PortHandler.initialize = function () {
@@ -66,7 +32,41 @@ PortHandler.initialize = function () {
 
     const self = this;
 
-    mdnsBrowser?.on('down', function (service) {
+    const http = require("http");
+    const bonjour = require('bonjour')();
+    
+    const MDNS_INTERVAL = 10000;
+    const TCP_CHECK_INTERVAL = 5000;
+    const TCP_TIMEOUT = 2000;
+    
+    if (GUI.isCordova()) {
+        self.mdnsBrowser = {
+            services: [],
+        };
+    
+        const zeroconf = cordova.plugins.zeroconf;
+        zeroconf.registerAddressFamily = 'ipv4'; // or 'ipv6' ('any' by default)
+        zeroconf.watchAddressFamily = 'ipv4'; // or 'ipv6' ('any' by default)
+        zeroconf.watch("_http._tcp.", "local.").subscribe(result => {
+            const action = result.action;
+            const service = result.service;
+            if (['added', 'resolved'].includes(action)) {
+                console.log("Zeroconf Service Changed", service);
+                self.mdnsBrowser.services.push({
+                    addresses: service.ipv4Addresses,
+                    txt: service.txtRecord,
+                    fqdn: service.hostname,
+                });
+            } else {
+                console.log("Zeroconf Service Removed", service);
+                self.mdnsBrowser.services = self.mdnsBrowser.services.filter(s => s.fqdn !== service.hostname);
+            }
+        });
+    } else {
+        bonjour.find({ type: 'http' });
+    }
+
+    self.mdnsBrowser?.on('down', function (service) {
         console.log('Lost an HTTP server:', service);
         let currentPorts = [];
         if (self.initialPorts && self.initialPorts.length > 0) {
@@ -86,10 +86,10 @@ PortHandler.initialize = function () {
                 const tcpPorts = self.initialPorts.filter(p => p.path.startsWith('tcp://'));
                 tcpPorts.forEach(function (port) {
                     const removePort = () => {
-                        mdnsBrowser?._removeService(port.fqdn);
+                        self.mdnsBrowser?._removeService(port.fqdn);
 
                         if (GUI.isCordova()) {
-                            mdnsBrowser.services = mdnsBrowser.services.filter(s => s.fqdn !== port.fqdn);
+                            self.mdnsBrowser.services = self.mdnsBrowser.services.filter(s => s.fqdn !== port.fqdn);
                         }
                     };
                     http.get({
@@ -123,7 +123,7 @@ PortHandler.initialize = function () {
 
     self.mdns_timer = setInterval(() => {
         if (!GUI.connected_to) {
-            mdnsBrowser?.update();
+            self.mdnsBrowser?.update();
         }
     }, MDNS_INTERVAL);
 
@@ -158,7 +158,7 @@ PortHandler.check_serial_devices = function () {
 
         let currentPorts = [
             ...cp,
-            ...(mdnsBrowser?.services?.filter(s => s.txt.vendor === 'elrs' && s.txt.type === 'rx')
+            ...(self.mdnsBrowser?.services?.filter(s => s.txt.vendor === 'elrs' && s.txt.type === 'rx')
                 .map(s => s.addresses.map(a => ({
                     path: `tcp://${a}`,
                     displayName: `${s.txt.target} - ${s.txt.version}`,
```